src/Core/Geometry/Volume.hpp

```diff
@@ -186,6 +186,10 @@ class RA_CORE_API VolumeGrid : public AbstractDiscreteVolume
 
     using Container = std::vector<ValueType>;
 
+    // For now, assume that AbstractDiscreteVolume::ValueType is Scalar
+    // vector4 : gradient and density
+    using Gradientcontainer = std::vector<Vector4>;
+
```

Comment:
Not sure to understand the comment (i.e. for now)
Should it state that ValueType is expected to be scalar, so the gradient is Ra::Vector<ValueType> or something like that ?

Reply:
For  now (the current implementation), the gradient storing and computation is only compatible with ValueType=Scalar. This is similar to the general note about volume stating that (file Volume.hpp, line 21): 
```
\note Right now the encoded function \f$f\f$ is dimension 1, ie. \f$f:\Re^3->\Re\f$. If needed,
 *          the class can be templated so the built-in typedef #ValueType is replaced by a
 * parameter.
```
This comment is just here to recall that if we change `AbstractDiscreteVolume::ValueType`, then we need to integrate this change here. May be reworded into
```suggestion
    // Right now, assume that AbstractDiscreteVolume::ValueType is Scalar
    // In this implementation, the type Vector4 store the gradient (x, y, z) and density (w)
    using Gradientcontainer = std::vector<Vector4>;

```